pkg/generate/dockercompose/generate.go

```diff
@@ -0,0 +1,562 @@
+					}
+					for _, path := range at {
+						c.VolumeMounts = append(c.VolumeMounts, kapi.VolumeMount{Name: name, MountPath: path})
+					}
+				}
+			}
+
+			pipeline, err := app.NewPipelineBuilder(k, nil, true).To(k).NewImagePipeline(k, inputImage)
+			if err != nil {
+				errs = append(errs, err)
+				break
+			}
+
+			if err := pipeline.NeedsDeployment(nil, nil, false); err != nil {
+				return nil, err
+			}
+
+			group = append(group, pipeline)
+		}
+		if err := group.Reduce(); err != nil {
+			return nil, err
+		}
+		pipelines = append(pipelines, group...)
+	}
+
+	if len(errs) > 0 {
+		return nil, utilerrs.NewAggregate(errs)
+	}
+
+	acceptors := app.Acceptors{app.NewAcceptUnique(kapi.Scheme), app.AcceptNew}
+	objects := app.Objects{}
+	accept := app.NewAcceptFirst()
+	for _, p := range pipelines {
+		accepted, err := p.Objects(accept, acceptors)
+		if err != nil {
+			return nil, fmt.Errorf("can't setup %q: %v", p.From, err)
+		}
+		objects = append(objects, accepted...)
+	}
+
+	// create services for each object with a name based on alias.
+	containers := make(map[string]*kapi.Container)
+	var services []*kapi.Service
+	for _, obj := range objects {
+		switch t := obj.(type) {
+		case *deployapi.DeploymentConfig:
+			ports := app.UniqueContainerToServicePorts(app.AllContainerPorts(t.Spec.Template.Spec.Containers...))
+			if len(ports) == 0 {
+				continue
+			}
+			svc := app.GenerateService(t.ObjectMeta, t.Spec.Selector)
+			if aliases[svc.Name].Len() == 1 {
+				svc.Name = aliases[svc.Name].List()[0]
+			}
+			svc.Spec.Ports = ports
+			services = append(services, svc)
+
+			// take a reference to each container
+			for i := range t.Spec.Template.Spec.Containers {
+				c := &t.Spec.Template.Spec.Containers[i]
+				containers[c.Name] = c
+			}
+		}
+	}
+	for _, svc := range services {
+		objects = append(objects, svc)
+	}
+
+	// for each container that defines VolumesFrom, copy equivalent mounts.
+	// TODO: ensure mount names are unique?
+	for target, otherContainers := range volumesFrom {
+		for _, from := range otherContainers {
+			for _, volume := range containers[from].VolumeMounts {
+				containers[target].VolumeMounts = append(containers[target].VolumeMounts, volume)
+			}
+		}
+	}
+
+	template.Objects = objects
+
+	// generate warnings
+	if len(warnings) > 0 {
+		allWarnings := sets.NewString()
+		for msg, services := range warnings {
+			allWarnings.Insert(fmt.Sprintf("%s: %s", strings.Join(services, ","), msg))
+		}
+		if template.Annotations == nil {
+			template.Annotations = make(map[string]string)
+		}
+		template.Annotations[app.GenerationWarningAnnotation] = fmt.Sprintf("not all docker-compose fields were honored:\n* %s", strings.Join(allWarnings.List(), "\n* "))
+	}
+
+	return template, nil
+}
+
+// extractFirstPorts converts a Docker compose port spec (CONTAINER, HOST:CONTAINER, or
+// IP:HOST:CONTAINER) to the first container and host port in the range.  Host port will
+// default to container port.
```

Comment:
since you're combining containers into a single pod when they share volumes, is there a risk of port conflicts if two images use the same port and share a volume?